client/file-transfer-service/src/commands.rs

```diff
@@ -61,6 +67,10 @@ pub enum RequestError {
     /// File not found in the registry.
     #[error("File not found in registry")]
     FileNotRegistered,
+    #[error("DownloadRequestId not found in internal mapping")]
+    DownloadRequestIdNotFound,
+    #[error("Failed to return response")]
+    ResponseFailure(OutgoingResponse),
```

Comment:
This only happens when sending the download response, right? I would rename this to `DownloadResponseFailure`, but we will probably make this specific to the command when we either have another command with this case or when refactoring errors.

Reply:
I added the `Download` prefix to the error and also made it return the `OutgoingResponse` struct in the error message. Will add a `TODO` so we can eventually refactor this and split into different enums